p2p/host/relay/relay.go

```diff
@@ -20,7 +20,31 @@ func Advertise(ctx context.Context, advertise discovery.Advertiser) {
 	go func() {
 		select {
 		case <-time.After(AdvertiseBootDelay):
-			discovery.Advertise(ctx, advertise, RelayRendezvous, discovery.TTL(AdvertiseTTL))
+			go func() {
```

Comment:
This is brought in from go-libp2p-discovery.

If it's better to use the Util Functions from there I can swtich back to using them.

Reply:
Well, those functions exist so we don't need to duplicate this code. What was the motivation for this change?

Reply:
Ah... I see.